lib/compilation_engine.rb

```diff
@@ -270,12 +270,15 @@ def compile_expression
   def compile_expression_list
     @output.puts("<expressionList>")
 
+    @expressions_count = 0
```

Comment:
Again, this instance variable approach is an okay hack for `Seven`, but it’s not going to work in general because subroutine calls can have other subroutine calls as arguments. In other words: `#compile_expression_list` calls `#compile expression`, which calls `#compile_term`, which can call `#compile_expression_list` again, and that’ll overwrite the previous value of `@expressions_count`. We need to do the AST-walking thing to make this work correctly.